### ISSUE TITLE
[Activation/Regulation functions not working properly]

### ISSUE THREAD

**Issue by Seledriac** [Seledriac/A-small-pedagogic-python-library-for-supervised-neural-networks#3]
**Describe the bug**
ReLu doesn't work without softmax and tanh is very unstable. Regarding regulation functions : softmax and normalization have exploding or vanishing output values problems. Overall, with regulation functions, the results are very bad.

**To Reproduce**
1. Create a model with ReLu or tanh and add a regulatino function
2. Make the model fit a training data with the SGD method
3. The results are always bad

**Expected behavior**
The only currently working combination is sigmoid with no regulation. This shouldn't be the case.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Python version: [e.g. 3.8.3
